File: ruby/testcode/hardcodedcreds_013.rb
require_relative 'shared'

# vuln-code-snippet start ruby_hardcoded_smtp
def send_email(req)
  smtp_config = { address: 'smtp.example.com', user_name: 'admin', password: 'P@ssw0rd123!' } # vuln-code-snippet vuln-line ruby_hardcoded_smtp
  BenchmarkResponse.ok("email sent via #{smtp_config[:address]}")
end
# vuln-code-snippet end ruby_hardcoded_smtp

File: rust/testcode/hardcodedcreds_013.rs
//! CWE-798: AWS credentials resolved via SDK default credential chain at runtime.

// vuln-code-snippet start testcodeHardcodedcreds013
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let bucket = req.param("bucket");
    // Simulates: aws_config::load_defaults(BehaviorVersion::latest()).await
    let credentials = load_aws_defaults(); // vuln-code-snippet target-line testcodeHardcodedcreds013
    let result = format!("S3 access to {} with default chain credentials", bucket);
    super::shared::BenchmarkResponse::ok(&result)
}

fn load_aws_defaults() -> String {
    // Simulates: aws_config::load_defaults(BehaviorVersion::latest()).await
    "default_chain".to_string()
}
// vuln-code-snippet end testcodeHardcodedcreds013
